src/modules/token/fungible/hooks/queries/useTokenBalances.js

```diff
@@ -21,7 +21,7 @@ export const useTokenBalances = ({
   client = defaultClient,
 } = {}) => {
   const [currentAccount] = useCurrentAccount();
-  const { address } = customConfig.params?.address || currentAccount.metadata || {};
+  const address = customConfig.params?.address || currentAccount.metadata?.address;
```

Comment:
Nice! This will also fix register as a validator! 👍 and probably some other bugs as well 🎉